reactor-core/src/main/java/reactor/util/context/Context.java

```diff
@@ -72,6 +95,10 @@ static Context of(Object key, Object value) {
 	 */
 	static Context of(Object key1, Object value1,
 			Object key2, Object value2) {
+		//prefer direct check for the simple 2 keys case over checkDuplicateKeys (saves up 1 array allocation)
+		if (key1.equals(key2)) {
+			throw new IllegalArgumentException("Found duplicate key in Context.of() with 2 key-value pairs: " + key1);
```

Comment:
Since the stracktrace will have `Context.of` already, maybe we should simplify the message to something like `"Duplicated key: " + key1`?

Reply:
I changed the message to the format `Key #keyN (valueN) is duplicated` across the board